src/ExadelMentorship.Persistence/WeatherHistorySavingRepository.cs

```diff
@@ -0,0 +1,31 @@
+﻿using ExadelMentorship.BusinessLogic.Interfaces;
+using ExadelMentorship.DataAccess;
+using ExadelMentorship.DataAccess.Entities;
+using ExadelMentorship.Persistence.Context;
+
+namespace ExadelMentorship.Persistence
+{
+    public class WeatherHistorySavingRepository : IWeatherHistorySavingRepository
+    {
+        private readonly ApplicationDbContext _dbContext;
+        private readonly IWeatherApiService _weatherApiService;
+
+        public WeatherHistorySavingRepository(ApplicationDbContext context, IWeatherApiService weatherApiService)
```

Comment:
It is better if repository would not depend on anything than DbContext and some other services like logger.

Reply:
done